Add documentation on how to run a Hackathon (#3202)
mattermost/docs: source/process/how-to-run-a-hackathon.rst

How to run a Hackathon
=========================

> This page helps outline best practices and recommended steps for running a Hackathon.

Reviewer: ```suggestion
This page outlines best practices and recommended steps for running a Hackathon.
```

Previously, Hackathons have run twice a year with one coinciding with MatterCon. In future, Hackathons may be run 3 or 4 times a year.

> T-minus counts measure "working days" (weekdays other than major holidays concurrent in US and Canada) prior to the Hackathon day. All times are based on 8:00am ([Palo Alto Time](http://everytimezone.com/)) unless otherwise stated.

Reviewer: Wrong link formatting.

A. (T-minus 60 working days)
----------------------------------------------------------------

1. Decide dates for the Hackathon. Some things to take into consideration:

  - Avoid overlap with major holidays in US, Canada, Europe or Asia (including Australia and Japan).

>   - Avoid overlap with significant release dates, such as release testing or feature complete. Recommended time is starting a Hackathon within a week after major feature complete.

Reviewer: ```suggestion
  - Avoid overlap with significant release dates, such as release testing or feature complete. Recommended time is starting a Hackathon within a week after feature complete.
```

  - Previous Hackathons ran for 2 days with presentations on the third date. Last time developers suggested adding a third date to allow more time for polish and increased quality.

2. Identify at least 1 or 2 developers to partner with during the Hackathon. Start by reaching out to either VP Engineering or Dev Leads to hear their recommendations.

B. (T-minus 40 working days)
----------------------------------------------------------------

1. Put together a project plan, including task description, start date, due date, owner and any additional notes. `See an example <https://docs.google.com/spreadsheets/d/15iiLPzJjlajHQVgQa9HulW9WfBauIlnkCOJ07oKz7_U>`_.
2. Identify 1 or 2 members in marketing to partner with, who can help with tweets, blog posts and other marketing content to generate awareness of the Hackathon.
3. Identify 1 or 2 members in ops to partner with, who can help with meeting scheduling, SWAG and other operational tasks.

C. (T-minus 30 working days)
----------------------------------------------------------------

1. Work with ops to decide on what SWAG to send for participation - both for contributors and staff.

> 2. Create a landing page for the Hackathon, which contains all information and to which other communication links to.

Reviewer: ```suggestion
2. Create a landing page for the Hackathon, which contains all information and to which other communication items are linked to.
```

  - This can be a webpage such as `one for Feb/2019 Hackfest <https://mattermost.com/hackfest2019/>`_ or a GitHub repository such as `one for Nov/2019 Hackathon <https://github.com/mattermost/mattermost-hackathon-nov2019>`_.
  - All processes and documentations should be clearly included on this landing page. This should include process for finding and assigning teams, and submission guidelines. FAQs can be incorporated as needed.

3. Create a GitHub repository for submitting Hackathon projects. This can be the same as the landing page if a GitHub repository is used for it.

D. (T-minus 20 working days)
----------------------------------------------------------------

1. Announce the Hackathon internally by posting in `Private Staff Discussion channel <https://community.mattermost.com/private-core/channels/platform>`_ and sharing on R&D weekly meeting.
2. Schedule a kick-off call with all Hackathon stakeholders, including the developer, marketing and ops members identified previously.

> 3. Define and set up metrics to measure impact of the Hackathon. Examples include: number of participants (staff + community separately), number of submissions, number of submissions shipped as a core feature, or as a plugin in the marketplace.

Reviewer: ```suggestion
3. Define and set up metrics to measure impact of the Hackathon. Examples include: number of participants (staff + community separately), number of submissions, number of submissions shipped as a core feature or as a plugin in the marketplace.
```

E. (T-minus 15 working days)
----------------------------------------------------------------

1. Create Meetup links for Hackathon idea presentation, kick-off and demos. For examples, see `Mattermost Meetup page <https://www.meetup.com/mattermost/>`_.
2. Share the Hackathon externally via Reddit, GitHub, Discourse and the `community server <https://community.mattermost.com>`_. For examples, see previous `Reddit post <https://www.reddit.com/r/Mattermost/comments/dvvsm4/mattermost_hackathon_nov_22_26/>`_, and `GitHub issue <https://github.com/mattermost/mattermost-server/issues/13087>`_.

> 3. Publish a blog post announcing the Hackathon to the Mattermost community, with links back to landing page for details on how to participate.

Reviewer: ```suggestion
3. Publish a blog post announcing the Hackathon to the Mattermost community, which links back to the landing page for details on how to participate.
```

F. (T-minus 10 working days)
----------------------------------------------------------------

1. Schedule a series of tweets announcing and reminding community of the upcoming Hackathon. See `this Google Doc <https://docs.google.com/document/d/1W5j5dYa3ZjSgEocniWXTEO2xjj5A8S9ihD03snUnLnA/edit#heading=h.s3nujhxhjrxo>`_ for examples.
2. Work with Ops team to get SWAG prizes ready and set up with Printfection.

G. (T-minus 5 working days)
----------------------------------------------------------------

> 1. Hold a Hackathon idea presentation call, scheduled for 60 minutes, with an additional 30 minutes for extra time. This time is used to gather feedback about ideas and attract others to work as a team. To see previous slides and meeting recording, see `Hackathon Nov/2019 Google Drive <https://drive.google.com/drive/folders/1iBVGDzdwZ72oEY6mshzjcgJQ4NdhXRfW>`_.

Reviewer: ```suggestion
1. Hold a Hackathon idea presentation call, scheduled for 60 minutes, with an additional 30 minutes for extra time. This time is used to gather feedback about ideas and to attract others to work as a team. To see previous slides and meeting recording, see `Hackathon Nov/2019 Google Drive <https://drive.google.com/drive/folders/1iBVGDzdwZ72oEY6mshzjcgJQ4NdhXRfW>`_.
```

H. (T-minus 0 working days)
----------------------------------------------------------------

1. Hold a Hackathon kick-off call, scheduled for 30 minutes. To see previous slides and meeting recording, see `Hackathon Nov/2019 Google Drive <https://drive.google.com/drive/folders/1iBVGDzdwZ72oEY6mshzjcgJQ4NdhXRfW>`_.

I. (T-plus 3 working days)
----------------------------------------------------------------

1. Hold a Hackathon demo call, scheduled for 2 hours, with an additional hour for extra time. This time is used for participants to present their project. Invite the entire Mattermost company to the call as optional attendee.

  - Provide guidelines and best practices for creating presentations (both for staff and community).
  - Three minutes allowed per Hackathon participant. For multiple submissions, five minutes total is allowed to present.

>   - Ask participants who can't attend submit a demo video, which are shared during the call.

Reviewer: ```suggestion
  - Ask participants who can't attend to submit a demo video, which are shared during the call.
```

J. (T-plus 5 working days)
----------------------------------------------------------------

1. Ask developers to hold a Hackathon retrospective and share feedback for future improvements.
2. Hold a Hackathon post-mortem call among key stakeholders (developers, marketing, ops).
3. Work with ops to gather a list of winners and send SWAG prizes via Printfection.
4. Gather a list of submissions and separate them by R&D feature teams. Product Managers on the respective teams will be responsible for discussing with Dev Leads on which submissions make sense to ship as core features or plugins, and scope out remaining work.

K. (T-plus 7 working days)
----------------------------------------------------------------

1. Publish blog post sharing insights of the Hackathon.
2. Update this process doc for changes in running a Hackathon.

Tips and Best Practices
----------------------------------------------------------------

1. Do not use your own Zoom link for Hackathon calls. Recommend using a company Zoom link, or one created specifically for Hackathon.

> 2. In Zoom, go to **Settings > Recording** and set **Recording consent** to true. This prompts participants for consent to be recorded when recording starts.

Reviewer: ```suggestion
2. In Zoom, go to **Settings > Recording** and set **Recording consent** to true. This prompts participants to consent to be recorded when recording starts.
```
